src/cpp/src/logit_processor.hpp

```diff
@@ -286,14 +329,15 @@ class LogitProcessor {
                 if (sampling_params.top_p != 1.0f) {
                     m_logit_transformers.emplace_back(new LogitTransformers::TopPFilter(sampling_params.top_p));
                 }
-                if (sampling_params.top_k > 0) {
+                // TODO: Uncommenting below condition requires changes in reference texts in tests
+                if (sampling_params.top_k > 0 /* && sampling_params.top_k < std::numeric_limits<size_t>::max() */) {
```

Comment:
I encountered an issue with exactly same line. Random sampling results seems to depend on logits order. TopK filter with default topK (`std::numeric_limits<size_t>::max()`) gives different results than removed filter.

Reply:
Exactly, currently default top_k results in TopK filter being applied always for multinomial sampling and logits always get sorted which impacts the output.

Reply:
Please uncomment and update a ref

Reply:
Agreed to do it in next PR